app/src/main/java/uk/gov/android/securestore/authentication/AuthenticatorCallbackHandler.kt

```diff
@@ -0,0 +1,10 @@
+package uk.gov.android.securestore.authentication
+
+data class AuthenticatorCallbackHandler(
+    val onSuccess: () -> Unit = {},
+    val onError: (
+        errorCode: Int,
+        errString: CharSequence
+    ) -> Unit = { _, _ -> },
+    val onFailure: () -> Unit = {}
+)
```

Comment:
If we're only passing in higher-order functions, are we happy with doing so or would we want an interface that instead matches these contractual obligations?

Reply:
I am happy with these for now, especially given that the actual usage of this stuff is 'internal' to the `SecureStore` workings